ibc/src/component/channel/stateful/proof_verification.rs

```diff
@@ -0,0 +1,306 @@
+use super::super::*;
+use ibc::clients::ics07_tendermint::client_state::ClientState as TendermintClientState;
+use ibc::core::ics04_channel::context::calculate_block_delay;
+use ibc::core::ics23_commitment::commitment::CommitmentPrefix;
+use ibc::core::ics23_commitment::commitment::CommitmentProofBytes;
+use ibc::core::ics23_commitment::commitment::CommitmentRoot;
+use ibc::core::ics23_commitment::merkle::apply_prefix;
+use ibc::core::ics23_commitment::merkle::MerkleProof;
+use ibc::core::ics23_commitment::specs::ProofSpecs;
+use ibc::core::ics24_host::identifier::ClientId;
+use ibc::core::ics24_host::path::AcksPath;
+use ibc::core::ics24_host::path::CommitmentsPath;
+use ibc::core::ics24_host::path::ReceiptsPath;
+use ibc::core::ics24_host::path::SeqRecvsPath;
+use ibc::core::ics24_host::Path;
+use ibc::downcast;
+use ibc_proto::ibc::core::commitment::v1::MerkleProof as RawMerkleProof;
+use prost::Message;
+
+use ibc::core::ics02_client::client_state::AnyClientState;
+use penumbra_chain::View as _;
+use sha2::{Digest, Sha256};
+
+// NOTE: this is underspecified.
```

Comment:
Should we keep a list of things that are underspecified, so we can contribute them to spec improvements?